File: scalechain-chain/src/test/scala/io/scalechain/blockchain/chain/TransactionAnalyzerSpec.scala
package io.scalechain.blockchain.chain

import io.scalechain.blockchain.transaction.TransactionTestDataTrait
import org.scalatest._

class TransactionAnalyzerSpec extends FlatSpec with BeforeAndAfterEach with TransactionTestDataTrait with Matchers {
  "sumAmount" should "should sum the amount of transaction outputs" in {
    TransactionAnalyzer.sumAmount(List(OUTPUT1, OUTPUT2, OUTPUT3)) shouldBe (OUTPUT1.value + OUTPUT2.value + OUTPUT3.value)
  }

  "calculateFee" should "" ignore {
    // TODO : Implement
  }

  "getSpentOutputs" should "" ignore {
    // TODO : Implement
  }
}

File: scalechain-chain/src/test/kotlin/io/scalechain/blockchain/chain/TransactionAnalyzerSpec.kt
package io.scalechain.blockchain.chain

import io.kotlintest.KTestJUnitRunner
import io.kotlintest.matchers.Matchers
import io.kotlintest.specs.FlatSpec
import io.scalechain.blockchain.transaction.TransactionTestData
import io.scalechain.blockchain.transaction.TransactionTestInterface
import org.junit.runner.RunWith
import java.math.BigDecimal

@RunWith(KTestJUnitRunner::class)
class TransactionAnalyzerSpec : FlatSpec(), TransactionTestInterface, Matchers {
  val D = TransactionTestData
  init {
    "sumAmount" should "should sum the amount of transaction outputs" {
      TransactionAnalyzer.sumAmount(listOf(D.OUTPUT1, D.OUTPUT2, D.OUTPUT3)) shouldBe BigDecimal(D.OUTPUT1.value + D.OUTPUT2.value + D.OUTPUT3.value)
    }

    /*
    "calculateFee" should "" {
      // TODO : Implement
    }

    "getSpentOutputs" should "" {
      // TODO : Implement
    }
    */
  }
}